lib/DxcSupport/HLSLOptions.cpp

```diff
@@ -69,6 +69,119 @@ const OptTable * hlsl::options::getHlslOptTable() {
   return g_HlslOptTable;
 }
 
+static std::vector<ArgPair> ComputeArgPairsFromArgList(const llvm::opt::InputArgList &argList) {
+  std::vector<ArgPair> ret;
+
+  for (llvm::opt::Arg *arg : argList) {
+    llvm::StringRef name = arg->getOption().getName();
+    llvm::StringRef value;
+
+    llvm::SmallString<64> argumentStorage; // Buffer for when we need to concatenate things.
+
+    // The input argument should be excluded from this list. All the uses for
+    // the ArgPair list used to manually filter out the input file. Do it here
+    // in the source instead.
+    if (arg->getOption().getKind() == llvm::opt::Option::InputClass) {
+      continue;
+    }
+    // This is a flag without value, like -Zi, or -Od
+    else if (arg->getOption().getKind() == llvm::opt::Option::FlagClass) {
+      assert(arg->getNumValues() == 0);
```

Comment:
nit: since the body of this condition is just an assert, we could hoist the assert out before the first `if` and have the assert contain the condition and the assert check, then this `else if` block could disappear.